ext/sqlite3/sqlite3.c

```diff
@@ -560,15 +559,6 @@ PHP_METHOD(SQLite3, query)
 		RETURN_FALSE;
 	}
 
-	/* If there was no return value then just execute the query */
-	if (!USED_RET()) {
-		if (sqlite3_exec(db_obj->db, ZSTR_VAL(sql), NULL, NULL, &errtext) != SQLITE_OK) {
-			php_sqlite3_error(db_obj, "%s", errtext);
-			sqlite3_free(errtext);
-		}
-		return;
```

Comment:
I don't think we want to drop this case. You can simply use a return value that's part of the valid set, so that would be `RETURN_FALSE` in this case. The actual value doesn't matter in that it is unobservable.

Reply:
> The actual value doesn't matter in that it is unobservable.

Ah, that's the piece I missed! And I believe this change will make the tests pass, thanks!

Reply:
I've just reverted this change, but to be honest, I wasn't sure what to do with the other similar changes (so I left them as-is for now).

Reply:
I think the other ones should be changed in the same way. RETURN_FALSE should work for them as well.

Reply:
Fixed - actually I stashed these changes yesterday :)